pkgs/build-support/php/hooks/composer-setup-hook.sh

```diff
@@ -0,0 +1,76 @@
+declare -a composerLock
+declare -a composerHome
+declare -a installComposerDevDependencies
+declare -a runComposerScripts
+declare -a runComposerPlugins
+declare -a dontComposerSetupConfigureHook
+declare -a dontComposerSetupBuildHook
+declare -a dontComposerSetupInstallHook
+declare -a version
+
+if [[ -z "${dontComposerSetupConfigureHook-}" ]]; then
+  postConfigureHooks+=(composerSetupConfigureHook)
+fi
+
+if [[ -z "${dontComposerSetupBuildHook-}" ]]; then
+  preBuildHooks+=(composerSetupBuildHook)
+fi
+
+if [[ -z "${dontComposerSetupInstallHook-}" ]]; then
+  preInstallHooks+=(composerSetupInstallHook)
+fi
+
+composerSetupConfigureHook() {
+    echo "Executing composerSetupConfigureHook"
+
+    if [[ ! -d ".composer" ]]; then
+        mkdir -p ".composer"
+    fi
+
+    if [[ -e "$composerLock" ]]; then
+        cp $composerLock composer.lock
+    fi
+
+    if [[ ! -f "composer.lock" ]]; then
+        echo "No composer.lock file found"
+        exit 1
+    fi
+
+    echo "Finished composerSetupConfigureHook"
+}
+
+composerSetupBuildHook() {
+    echo "Executing composerSetupBuildHook"
+
+    argstr=("--no-interaction" "--download-only")
```

Comment:
Taking a closer look I'm not sure that `--download-only` is good enough to be used a FOD: it will pull tarballs directly from the GitHub API to put it in the cache. The GitHub API does not provide stable tarballs: https://github.blog/2023-02-21-update-on-the-future-stability-of-source-code-archives-and-hashes/

I see 2 possible ways forward:
* decompressing and recompressing every files the cache in a way that is reproducible
* not relying on the cache: the FOD would be the `vendor/` directory generated with `composer install --no-autoloader && rm vendor/composer/InstalledVersions.php` then as a second phase copying back `InstalledVersions.php` from Composer sources and generating the autoloader with `composer dumpautoload`


Composer odes not make things easy here :/

Reply:
I will make some tests today, that makes more sense than dealing with cache and archives.